test/python/quantum_info/operators/symplectic/test_pauli_table.py

```diff
@@ -0,0 +1,1240 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=invalid-name
+
+"""Tests for PauliTable class."""
+
+import unittest
+from test import combine
+from ddt import ddt
+import numpy as np
+from scipy.sparse import csr_matrix
+
+from qiskit import QiskitError
+from qiskit.test import QiskitTestCase
+from qiskit.quantum_info.operators.symplectic import PauliTable
+
+
+def pauli_mat(label):
+    """Return Pauli matrix from a Pauli label"""
+    mat = np.eye(1, dtype=complex)
+    for i in label:
+        if i == 'I':
+            mat = np.kron(mat, np.eye(2, dtype=complex))
+        elif i == 'X':
+            mat = np.kron(mat, np.array([[0, 1], [1, 0]], dtype=complex))
+        elif i == 'Y':
+            mat = np.kron(mat, np.array([[0, -1j], [1j, 0]], dtype=complex))
+        elif i == 'Z':
+            mat = np.kron(mat, np.array([[1, 0], [0, -1]], dtype=complex))
+        else:
+            raise QiskitError('Invalid Pauli string {}'.format(i))
+    return mat
+
+
+class TestPauliTableInit(QiskitTestCase):
+    """Tests for PauliTable initialization."""
+
+    def test_array_init(self):
+        """Test array initialization."""
+        # Matrix array initialization
+        with self.subTest(msg='bool array'):
+            target = np.array([[False, False], [True, True]])
+            value = PauliTable(target)._array
+            self.assertTrue(np.all(value == target))
+
+        with self.subTest(msg='bool array no copy'):
+            target = np.array([[False, True], [True, True]])
+            value = PauliTable(target)._array
+            value[0, 0] = not value[0, 0]
+            self.assertTrue(np.all(value == target))
+
+        with self.subTest(msg='bool array raises'):
+            array = np.array([[False, False, False],
+                              [True, True, True]])
+            self.assertRaises(QiskitError, PauliTable, array)
+
+    def test_vector_init(self):
+        """Test vector initialization."""
+        # Vector array initialization
+        with self.subTest(msg='bool vector'):
+            target = np.array([False, False, False, False])
+            value = PauliTable(target)._array
+            self.assertTrue(np.all(value == target))
+
+        with self.subTest(msg='bool vector no copy'):
+            target = np.array([False, True, True, False])
+            value = PauliTable(target)._array
+            value[0, 0] = not value[0, 0]
+            self.assertTrue(np.all(value == target))
+
+    def test_string_init(self):
```

Comment:
I think I didn't use `@combine` or one of the decorators here because I would have had to do something like
```
@combine(pairs=[(str_label, target_array), ...])
```
which would have made the decorator very long.